config/kernel.m4

```diff
@@ -662,8 +662,10 @@ AC_DEFUN([ZFS_LINUX_COMPILE], [
 		build kernel modules with LLVM/CLANG toolchain])
 	AC_TRY_COMMAND([
 	    KBUILD_MODPOST_NOFINAL="$5" KBUILD_MODPOST_WARN="$6"
-	    make modules -k -j$TEST_JOBS ${KERNEL_CC:+CC=$KERNEL_CC} ${KERNEL_LD:+LD=$KERNEL_LD} ${KERNEL_LLVM:+LLVM=$KERNEL_LLVM} -C $LINUX_OBJ $ARCH_UM
-	    M=$PWD/$1 >$1/build.log 2>&1])
+	    make modules -k -j$TEST_JOBS ${KERNEL_CC:+CC=$KERNEL_CC}
+	    ${KERNEL_LD:+LD=$KERNEL_LD} ${KERNEL_LLVM:+LLVM=$KERNEL_LLVM}
+	    CONFIG_MODULES=y CFLAGS_MODULE=-DCONFIG_MODULES
```

Comment:
Good idea.  It'd also be nice to add a `CONFIG_MODULES` check to `config/kernel-config-defined.m4`.  We'd of course only want it to fail when `!CONFIG_MODULES` and `test "x$enable_linux_builtin" != xyes`.  This way we could print a clear error message with `AC_MSG_ERROR`.

Reply:
Indeed, I've planned to do that when I enforced `CONFIG_BLOCK`, I wanted to wait until I have more insights into the `CONFIG_MODULES` option.

On a side note, why
`"x$enable_linux_builtin" != xyes` vs. `"$enable_linux_builtin" != yes`
?

Reply:
Good question.  It's a long standing m4 convention to handle the case where the variable isn't defined for some reason.  By prefixing both sides with a constant 'x' you're guaranteed the expanded shell syntax will still be valid with an operand on each side of the operator.